nexus/tests/integration_tests/instances.rs

```diff
@@ -717,14 +717,64 @@ async fn test_instance_migrate_v2p(cptestctx: &ControlPlaneTestContext) {
     }
 }
 
+/// Assert values for fleet, silo, and project using both system and silo
+/// metrics endpoints
+async fn assert_metrics(
+    cptestctx: &ControlPlaneTestContext,
+    project_id: Uuid,
+    disk: i64,
+    cpus: i64,
+    ram: i64,
+) {
+    cptestctx.oximeter.force_collect().await;
+
+    for id in &[None, Some(project_id)] {
+        assert_eq!(
+            get_latest_silo_metric(
+                cptestctx,
+                "virtual_disk_space_provisioned",
+                *id,
+            )
+            .await,
+            disk
+        );
+        assert_eq!(
+            get_latest_silo_metric(cptestctx, "cpus_provisioned", *id).await,
+            cpus
+        );
+        assert_eq!(
+            get_latest_silo_metric(cptestctx, "ram_provisioned", *id).await,
+            ram
+        );
+    }
+    for id in &[None, Some(*SILO_ID)] {
+        assert_eq!(
+            get_latest_system_metric(
+                cptestctx,
+                "virtual_disk_space_provisioned",
+                *id
+            )
+            .await,
+            disk
+        );
+        assert_eq!(
+            get_latest_system_metric(cptestctx, "cpus_provisioned", *id).await,
+            cpus
+        );
+        assert_eq!(
+            get_latest_system_metric(cptestctx, "ram_provisioned", *id).await,
+            ram
+        );
+    }
+}
+
```

Comment:
now asserting about both metrics endpoints. extracted helper because the test would be ridiculously long otherwise